peps/pep-9999.rst

```diff
@@ -552,29 +553,71 @@ Each entry in this list is defined as a dictionary with these fields:
+      - ``dict[Literal['install', 'query'], dict[Literal['command', 'requires_elevation', 'multiple_specifiers'], list[str] | bool | Literal['always', 'name-only', 'never']]]``
+      - Commands used to install or query the given package(s). Only two keys
+        are allowed: ``install`` and ``query``. Their value is a dictionary
+        with:
+
+        - a required key ``command`` that takes a list of strings
+          (as expected by ``subprocess.run``).
+
+        - an optional ``requires_elevation`` boolean (``False`` by default)
+          to indicate whether the command must run with elevated permissions (
+          e.g. administrator on Windows, superuser on Linux and macOS).
+
+        - an enum ``multiple_specifiers`` that determines whether the command
+          accepts multiple package specifiers at the same time, accepting one of:
+
+            - ``always``, default in ``install``.
+
+            - ``name-only``, the command only accepts multiple specifiers if they do
+              not contain version constraints.
+
+            -  ``never``, default in ``query``.
+
+        Exactly one of the ``command`` items MUST include a ``{}`` placeholder,
+        which will be replaced by the mapped package identifier(s). The
+        ``install`` command SHOULD support the placeholder being replaced by
+        multiple identifiers, ``query`` MUST only receive a single identifier
+        per command.
+      - True
+    * - ``specifier_syntax``
+      - ``dict[Literal['name_only', 'exact_version', 'version_ranges'], None | list[str] | dict[Literal['and', 'equal', 'greater_than', 'greater_than_equal', 'less_than', 'less_than_equal', 'not_equal', 'syntax'], None | str | list[str]]``
+      - Mapping of allowed PEP440 version comparison operators to the syntax
+        used in this package manager. Three top-level keys are accepted:
+
+        - ``name_only`` takes a list of strings as the syntax used for specifiers
+          that do not contain any version information; it MUST include the placeholder
+          ``{name}``.
+
+        - ``exact_version`` takes a list of strings as the syntax used for
+          specifiers that only express an exact version constraints; it must
+          include the placeholders ``{name}`` and ``{version}``.
+
+        - ``version_ranges`` is a dictionary where:
+
+          - the key ``syntax`` takes a list of strings that MUST include the
+            ``{ranges}`` placeholder (to be replaced by the maybe-joined
+            version constraints, to be determined by the value of ``and``) and
+            MAY include the ``{name}`` placeholder.
+
+          - the keys ``equal``, ``greater_than``, ``greater_than_equal``,
+            ``less_than``, ``less_than_equal``, and ``not_equal`` take a string
+            that MUST include the ``{version}`` placeholder, and MAY include
+            ``{name}``, or ``None`` to indicate that the respective operator is
+            not supported.
```

Comment:
```suggestion
          - the keys ``equal``, ``greater_than``, ``greater_than_equal``,
            ``less_than``, ``less_than_equal``, and ``not_equal`` take a string
            if the operator is supported, ``None`` otherwise. In the former case,
            the value MUST include the ``{version}`` placeholder, and MAY include
            ``{name}``.
```

Otherwise it reads like "MAY include `{name}` or `None`".

Also, do we require all keys to be present, or do we assume that missing = None?

Reply:
See [3541839](https://github.com/rgommers/peps/pull/50/commits/3541839500f4ad5aed346a32f1c443abd583f61a) and [42fe19b](https://github.com/rgommers/peps/pull/50/commits/42fe19b4eb8207e109f6e0acefad070af59a2807)